src/main/java/rx/internal/schedulers/NewThreadWorker.java

```diff
@@ -50,6 +53,16 @@
         PURGE = new AtomicReference<ScheduledExecutorService>();
         PURGE_FORCE = Boolean.getBoolean(PURGE_FORCE_KEY);
         PURGE_FREQUENCY = Integer.getInteger(FREQUENCY_KEY, 1000);
+
+        // Cached to prevent method call
+        IS_ANDROID = PlatformDependent.isAndroid();
```

Comment:
I don't think so, we trying to enable cancel policy if current platform is not Android or if it's Android and API >= 21, that's why I have 2 separate booleans.

Or maybe I just don't see easier way.